liboverscroll/src/main/java/me/everything/android/ui/overscroll/adapters/ViewPagerOverScrollDecorAdapter.java

```diff
@@ -0,0 +1,69 @@
+package me.everything.android.ui.overscroll.adapters;
+
+import android.support.v4.view.ViewPager;
+import android.view.View;
+
+import me.everything.android.ui.overscroll.HorizontalOverScrollBounceEffectDecorator;
+
+/**
+ * Created by Bruce too
+ * Enhance by amit
+ * On 2016/6/16
+ * At 14:51
+ * An adapter to enable over-scrolling over object of {@link ViewPager}
+ *
+ * @see HorizontalOverScrollBounceEffectDecorator
+ */
+public class ViewPagerOverScrollDecorAdapter implements IOverScrollDecoratorAdapter, ViewPager.OnPageChangeListener {
+
+    protected final ViewPager mViewPager;
+
+    protected int mLastPagerPosition = 0;
+    protected float mLastPagerScrollOffset;
+
+    public ViewPagerOverScrollDecorAdapter(ViewPager viewPager) {
+        this.mViewPager = viewPager;
+
+        mViewPager.addOnPageChangeListener(this);
+
+        mLastPagerPosition = mViewPager.getCurrentItem();
+        mLastPagerScrollOffset = 0f;
+    }
+
+    @Override
+    public View getView() {
+        return mViewPager;
+    }
+
+    @Override
+    public boolean isInAbsoluteStart() {
+
+        return mLastPagerPosition == 0 &&
+                mLastPagerScrollOffset == 0f &&
+                !mViewPager.canScrollHorizontally(-1);
```

Comment:
Revisiting this, I now realize the last condition is unnecessary - at least I think so. Could you please try to check how well the pager works without out? I would GREATLY appreciate that!